raft/src/main/java/org/apache/kafka/raft/LeaderState.java

```diff
@@ -304,6 +305,12 @@ public String toString() {
         }
     }
 
+    @Override
+    public boolean grantVote(int candidateId, Supplier<Boolean> logComparator) {
+        log.debug("Rejecting vote request since we are already leader on that epoch");
```

Comment:
Maybe we can improve some of these log messages by adding the `candidateId`?

Reply:
Good suggestions!